pkg/util/sketch/cms_test.go

```diff
@@ -0,0 +1,52 @@
+//go:build sketch
+// +build sketch
+
+package sketch
+
+import (
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestCMS(t *testing.T) {
+	myCms, _ := NewCountMinSketch(10, 4)
+
+	k := 1
+	numStreams := 10
+	maxPerStream := 100
+	events := make([]event, 0)
+	max := int64(0)
+
+	for j := 0; j < numStreams-k; j++ {
+		num := int64(maxPerStream)
+		n := rand.Int63n(num) + 1
+		if n > max {
+			max = n
+		}
+		for z := 0; z < int(n); z++ {
+			events = append(events, event{name: strconv.Itoa(j), count: 1})
+		}
+		//total += n
+		//oTotal += int(n)
+	}
+	// then another set of things more than the max of the previous entries
+	for z := numStreams - k; z < numStreams; z++ {
+		n := int64(rand.Int63n(int64(maxPerStream)) + 1 + max)
+		for x := 0; x < int(n); x++ {
+			events = append(events, event{name: strconv.Itoa(z), count: 1})
+		}
+		//total += n
+		//oTotal += int(n)
+	}
+
+	rand.Seed(time.Now().UnixNano())
```

Comment:
The seed should be a constant in order to compare different runs.

Reply:
The amount of randomized events is already non-deterministic, and randomized # of events plus different sorting is a more realistic representation of data we might have to run a topk query against. We can have a separate test that has a static dataset and sorting.

Reply:
Which events do you mean? They are so random but the same for each run.

Reply:
no, for each run of this test each stream (by string identifier `0 - numStreams-1` has a random amount of event occurrences between 1 and 100, essentially 10 streams each with 1-100 log line entries